### PR TITLE
Add a new command 0x08

### PR DESCRIPTION
**Pull request by bubbledevteam**
Add a new command 0x08(implemented in Bubble FW v2.6 and upper) which request decrypted Libre2/Pro/US 14days raw data from Bubble, the data Structure is the same as Libre1,make xdrip+ algorithm work with all Libre type sensors.

### CODE REVIEW

**Comment by jamorham** [comment]
Hi thanks for this. There seem to be quite a few changes in the diff. Are there any unrelated changes to those that have already been described included in this pr?

**Comment by tzachi-dar** [comment]
Generally speaking, this PR does a few things that can be categorized as:
1) Allow bubble to decrypt libre2 data and send it to xDrip as libre1 data.
2) Some general changes to the code.

Right now I'm working on a similar pr that will allow all devices to work on libre2 data (https://github.com/NightscoutFoundation/xDrip/pull/1429).

So, I think that it would be better if this PR will be limited to the changes in the bubble code only.
On the HandleGoodReading a new parameter was added: decripted_data 

If the bubble fw has already decrypted the code, they can simply mark this data as decrypted, and all should just work.

**Comment by tolot27** [comment]
Does this PR requires #1396? If yes, can they be merged?

**Comment by jamorham** [comment]
What is the current state of this PR, is the command supported elsewhere already? Can it be cleaned up to just include the changes needed? thanks

**Comment by parapenT1sta** [comment]
@tolot27 and @jamorham Do you think this can be merge? This will be usefull for users using Bubble and not requiring OOP2 to be installed.

**Comment by parapenT1sta** [comment]
@jamorham I think this command is also used on Diabox.

**Comment by parapenT1sta** [comment]
@jamorham https://github.com/JohanDegraeve/xdripswift/blob/master/xdrip/BluetoothTransmitter/CGM/Libre/Bubble/CGMBubbleTransmitter.swift#L164

**Comment by tolot27** [comment]
@bubbledevteam Please can you resolve the merge conflicts and limit the changes to the bubble code only?

**Review by tzachi-dar — CHANGES_REQUESTED** [review]
I have read this pr again after oop2 code is already working.

This change (with the modifications asked) should allow people to use libre2 without oop2.

This will be super helpful for people with watches that are not supported by oop2.

File: app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/LibreUtils.java [hunk 1]
```diff
@@ -162,8 +162,21 @@ public static String decodeSerialNumber(byte[] input) {
             binS = String.format("%8s", Integer.toBinaryString(uuidShort[i] & 0xFF)).replace(' ', '0');
             binary += binS;
         }
-        
+
         String v = "0";
+        byte[] patchInfo = PersistentStore.getBytes("patchInfo");
```
[the remark this answers is not in the document]
Comment (bubbledevteam) [reply app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/LibreUtils.java:167]: this is for distinguishing the sensor type, I tested the previous codes ,the sensor type is not right displayed in xdrip+.

And this is for Bubble only as I code it if the patchInfo is null, just make it Libre as default

File: app/src/main/java/com/eveningoutpost/dexdrip/LibreAlarmReceiver.java [hunk 2]
```diff
@@ -149,11 +149,21 @@ public void run() {
                                 Log.d(TAG, "At Start: Oldest : " + JoH.dateTimeText(oldest_cmp) + " Newest : " + JoH.dateTimeText(newest_cmp));
 
                                 final String data = bundle.getString("data");
+                                final String from = bundle.getString("from");
                                 final int bridge_battery = bundle.getInt("bridge_battery");
                                 if (bridge_battery > 0) {
                                     Pref.setInt("bridge_battery", bridge_battery);
                                     CheckBridgeBattery.checkBridgeBattery();
                                 }
+                                if ("Diabox".equals(from)) {
```
[the remark this answers is not in the document]
Comment (bubbledevteam) [reply app/src/main/java/com/eveningoutpost/dexdrip/LibreAlarmReceiver.java:158]: I  am sorry to submit this part ,in fact it is for this https://github.com/NightscoutFoundation/xDrip/pull/1395

**Pull request closed without merging**